src/index.js

```diff
@@ -1,52 +1,52 @@
 import React from "react"
 import PropTypes from "prop-types"
 
-function OutboundLink(props) {
+const NEWTAB = '_blank';
+const MIDDLE_CLICK = 1;
+
+function OutboundLink({ eventName, eventDetails, href, target, onClick, onMouseDown, children, ...rest }) {
+  const handleClick = e => {
+    const amplitudeEventName = eventName || window.amplitudeEventTypes.outboundLinkClick;
+    const amplitudeDetails = Object.assign({ href: href }, eventDetails)
+    const sameTarget = target !== NEWTAB;
+    const normalClick = !(e.ctrlKey || e.shiftKey || e.metaKey || e.button === MIDDLE_CLICK);
+
+    if (sameTarget && normalClick) {
+      e.preventDefault();
+      window.amplitude.getInstance().logEvent(amplitudeEventName, amplitudeDetails, () => {
+        window.location.href = href;
+      });
+    } else {
+      window.amplitude.getInstance().logEvent(amplitudeEventName, amplitudeDetails);
+    }
+
+    onClick && onClick(e);
+  };
+
+  // Adding mouseDown handler to track MIDDLE_CLICK, because MIDDLE_CLICK doesn't reach onClick event
+  const handleMouseDown = e => {
+    if (e.button === MIDDLE_CLICK) {
+      window.amplitude.getInstance().logEvent(amplitudeEventName, amplitudeDetails);
+    }
+
+    onMouseDown && onMouseDown(e);
+  };
+
   return (
-    <a
-      {...props}
-      onClick={e => {
-        if (typeof props.onClick === `function`) {
-          props.onClick()
-        }
-        let redirect = true
-        if (
-          e.button !== 0 ||
-          e.altKey ||
-          e.ctrlKey ||
-          e.metaKey ||
-          e.shiftKey ||
-          e.defaultPrevented
-        ) {
-          redirect = false
-        }
-        if (props.target && props.target.toLowerCase() !== `_self`) {
-          redirect = false
-        }
-        if (typeof window.amplitude === 'object') {
-          window.amplitude.getInstance().logEvent(window.amplitudeEventTypes.outboundLinkClick, {
-            href: props.href,
-          }, () => {
-            if (redirect) {
-              document.location = props.href
-            }
-          })
-        } else {
-          if (redirect) {
-            document.location = props.href
-          }
-        }
-
-        return false
-      }}
-    />
-  )
+    <a href={href} target={target} onClick={handleClick} onMouseDown={handleMouseDown} {...rest}>
+      {children}
+    </a>
+  );
 }
 
 OutboundLink.propTypes = {
+  eventName: Proptypes.string,
+  eventDetails: Proptypes.any,
```

Comment:
Can you rename these to eventType and eventProperties to match Amplitude's documentation/terminology for these?

Reply:
sure!